hydromt/components/mesh.py

```diff
@@ -11,33 +11,370 @@
+        write_optional_ugrid_attributes: bool = True,
+        **kwargs,
+    ) -> None:
+        """Write model grid data to a netCDF file at <root>/<fn>.
+
+        Keyword arguments are passed to :py:meth:`xarray.Dataset.to_netcdf`.
+
+        Parameters
+        ----------
+        fn : str, optional
+            Filename relative to the model root directory, by default 'grid/grid.nc'.
+        write_optional_ugrid_attributes : bool, optional
+            If True, write optional ugrid attributes to the netCDF file, by default
+            True.
+        **kwargs : dict
+            Additional keyword arguments to be passed to the
+            `xarray.Dataset.to_netcdf` method.
+        """
+        if self.data is None:
+            self._logger.debug("No mesh data found, skip writing.")
+            return
+        self._model_root._assert_write_mode()
+
+        # filename
+        _fn = join(self._model_root.path, fn)
+        if not isdir(dirname(_fn)):
+            os.makedirs(dirname(_fn))
+        self._logger.debug(f"Writing file {fn}")
+        ds_out = self.data.ugrid.to_dataset(
+            optional_attributes=write_optional_ugrid_attributes,
+        )
+        if self.crs is not None:
+            # save crs to spatial_ref coordinate
+            ds_out = ds_out.rio.write_crs(self.crs)
+        ds_out.to_netcdf(_fn, **kwargs)
+
+    def read(
+        self, fn: str = "mesh/mesh.nc", crs: Optional[Union[CRS, int]] = None, **kwargs
+    ) -> None:
+        """Read model mesh data at <root>/<fn> and add to mesh property.
+
+        key-word arguments are passed to :py:meth:`~hydromt.models.Model.read_nc`
+
+        Parameters
+        ----------
+        fn : str, optional
+            filename relative to model root, by default 'mesh/mesh.nc'
+        crs : CRS or int, optional
+            Coordinate Reference System (CRS) object or EPSG code representing the
+            spatial reference system of the mesh file. Only used if the CRS is not
+            found when reading the mesh file.
+        **kwargs : dict
+            Additional keyword arguments to be passed to the `read_nc` method.
+        """
+        self._model_root._assert_read_mode()
+        ds = xr.merge(
+            read_nc(
+                fn, root=self._model_root.path, logger=self._logger, **kwargs
+            ).values()
+        )
+        uds = xu.UgridDataset(ds)
+        if ds.rio.crs is not None:  # parse crs
+            uds.ugrid.set_crs(ds.raster.crs)
+            uds = uds.drop_vars(GEO_MAP_COORD, errors="ignore")
+        else:
+            if not crs:
+                raise ValueError(
+                    "no crs is found in the file nor passed to the reader."
+                )
+            else:
+                uds.ugrid.set_crs(crs)
+                self._logger.info(
+                    "no crs is found in the file, assigning from user input."
+                )
+        self._data = uds
+
+    def create(
```

Comment:
This is to create a 2D mesh so I would keep it in the function name. the current MeshComponent handles multiple grids which can be 1D or 2D or 3D.

```suggestion
    def create2d(
```